src/shared/ui/list/list.tsx

```diff
@@ -1,27 +1,46 @@
-// todo delete TextWithLink
+import { HTMLAttributes } from 'react';
+import { type VariantProps, cva } from 'class-variance-authority';
 import classNames from 'classnames/bind';
 import { TextWithLink } from '../text-with-link';
-import { LinkList } from '@/widgets/required/required.types';
+import { ListData } from '@/shared/types';
 
 import styles from './list.module.scss';
 
-export type ListProps = {
-  data: ListData;
-  marked?: boolean;
-};
-
-export type ListData = (string | LinkList)[] | [] | undefined;
+type ListProps = Pick<HTMLAttributes<HTMLElement>, 'className'>
+  & VariantProps<typeof listVariants>
+  & Record<'data', ListData>;
 
 export const cx = classNames.bind(styles);
 
-export const List = ({ data, marked = true }: ListProps) => {
+const listVariants = cva(cx('list'), {
+  variants: {
+    size: {
+      compact: cx('compact'),
+      medium: cx('medium'),
+    },
```

Comment:
Do we really need that prop? I can't find any usage of `compact`

Reply:
There's it (2nd item)
![image](https://github.com/user-attachments/assets/240fecb3-2771-40ee-a1ce-d11f3fb35b99)

Reply:
Why not use it there then?